src/main/kotlin/com/github/avrokotlin/avro4k/decoder/SealedClassDecoder.kt

```diff
@@ -27,8 +30,8 @@ class SealedClassDecoder (descriptor: SerialDescriptor,
    private var currentState = DecoderState.BEFORE
 
    var leafDescriptor : SerialDescriptor = descriptor.leavesOfSealedClasses().firstOrNull {
-      val schemaName = RecordNaming(value.schema.fullName, emptyList())
-      val serialName = RecordNaming(it)
+      val schemaName = RecordNaming(value.schema.fullName, emptyList(), DefaultNamingStrategy)
```

Comment:
Maybe it would be good to introduce a new invoke method on RecordNaming using the DefaultNamingStrategy. Just from reading this code one might get the impression that passing "DefaultNamingStrategy" at this point is incorrect because namingStrategy exists. But as the value.schema.fullName has already been decoded using the namingStrategy, passing the DefaultNamingStrategy is actually correct. Having a new invoke method without a naming strategy would imply that the name will not be changed.

Reply:
Added new invoke method. I wonder if it's a abit anti-pattern or should just default values be used. Or the whole RecordNaming class looks now a bit weird perhaps?